### PR TITLE
e2e: auth key put test

### PR DESCRIPTION
**Pull request by heyitsanthony**
/cc @mitake @xiang90

### CODE REVIEW

File: etcdserver/v3_server.go [hunk 1]
```diff
@@ -73,7 +73,7 @@ type Authenticator interface {
 }
 
 func (s *EtcdServer) Range(ctx context.Context, r *pb.RangeRequest) (*pb.RangeResponse, error) {
-	if r.Serializable {
+	if r.Serializable && s.AuthStore().IsRangePermitted(r.Header, string(r.Range.Key), string(r.RangeEnd)) {
```
Comment (mitake) [inline etcdserver/v3_server.go]: `r.Key` instead of `r.Range.Key`

Comment (mitake) [inline etcdserver/v3_server.go]: RequestHeader isn't prepared in this scope. It should be created from metadata associated with context. See https://github.com/coreos/etcd/blob/master/etcdserver/v3_server.go#L420

Comment (heyitsanthony) [inline etcdserver/v3_server.go]: yeah, I ran e2e but forgot to do a full build... all fixed

**Comment by xiang90** [comment]
lgtm. There are a few thing I want to clean up in etcdserver part though. I will do it in another pr.

**Pull request merged**